### DISCUSSION TITLE
Github action logs to sumo logic

### DISCUSSION THREAD

**Discussion by shaktikadapure** [community/community#26446] [Actions]
<p>I was evaluating github actions for CI/CD. One of the requirements is to push the loggings generated by github actions to sumo logic or some thing similar. Is there any kind of a proven methodology to do such a thing?</p>
<p>I would also like to know if the loggings can happen in sumo logic or any similar entity and not happen in github at all?</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Hi <a class="mention" href="/skadapureebsco">@skadapureebsco</a>,</p>
<p>I don’t know of anything currently available for sending logs to Sumo.</p>
<p>You might have to set something up manually using the Actions API. <a href="https://developer.github.com/v3/actions/workflow-runs/#list-workflow-run-logs" rel="nofollow noopener">https://developer.github.com/v3/actions/workflow-runs/#list-workflow-run-logs</a> </p>

**Comment by shaktikadapure** [comment]
<p>Thanks for the quick reply. I could accomplish this manually by getting the <em>logs.zip</em> from this end point - </p>
<p><a href="https://api.github.com/repos/mrOrg/myRepo/actions/runs/" class="inline-onebox-loading" rel="nofollow noopener">https://api.github.com/repos/mrOrg/myRepo/actions/runs/</a>&lt;&lt;run_id&gt;&gt;/logs. Then set up a collector in sumologic with http as a source. I followed these instructions- <a href="https://help.sumologic.com/03Send-Data/Setup-Wizard/Collect-from-Custom-Apps/Collect_Streaming_Data_from_HTTP" rel="nofollow noopener">https://help.sumologic.com/03Send-Data/Setup-Wizard/Collect-from-Custom-Apps/Collect_Streaming_Data_from_HTTP. </a>I unzipped the _logs.zip _and transferred file using cURL. </p>
<p>Only one additional thing I am looking for to automate this is to have these steps in the build itself and utilize environment variable _GITHUB_RUN_ID _to access the logs for current build. I introduced a 30s wait for the build and try to get the logs for current build on the browser and I get a 404. So, currently it seems only possible to get logs only after the build and not during. Is there a way the snapshot of logs are accessible during the build?</p>
